tmail-backend/guice/protocols/jmap/src/main/java/com/linagora/tmail/james/jmap/firebase/FirebaseModuleChooser.java

```diff
@@ -0,0 +1,24 @@
+package com.linagora.tmail.james.jmap.firebase;
+
+import java.util.List;
+
+import com.google.inject.Module;
+
+public class FirebaseModuleChooser {
+    public enum ServerType {
+        MEMORY,
+        DISTRIBUTED,
+    }
+
+    public static List<Module> chooseFirebase(FirebaseModuleChooserConfiguration moduleChooserConfiguration, ServerType serverType) {
+        if (moduleChooserConfiguration.enable()) {
+            return switch (serverType) {
+                case MEMORY -> List.of(new MemoryFirebaseSubscriptionRepository.Module(),
+                    new FirebaseCommonModule());
+                case DISTRIBUTED -> List.of(new CassandraFirebaseSubscriptionRepositoryModule(),
+                    new FirebaseCommonModule());
+            };
+        }
```

Comment:
Not clean from a maven dependency management perspective, as you would end up with all distributed dependencies on the memory classpath.

Here it is better to duplicate: have 3 module chooser.